File: lua/completion/test_prompts.lua
local M = {}
-- EXPECTED OUTPUT:
--         sum = sum + i
--         i = i + 1
--     end
--
--     return sum
-- end
function M.sum_upto(n)
    local sum = 0
    local i = 1

    while i <= n do
    end
end

-- EXPECTED OUTPUT:
--         return true
--     end
--
--     return false
-- end
function M.contains_zero(values)

    for i = 1, #values do
        if values[i] == 0 then
        end
    end
end

-- EXPECTED OUTPUT:
--             result[#result + 1] = values[i]
--         end
--     end
--
--     return result
-- end
function M.filter_positive(values)
    local result = {}

    for i = 1, #values do
        if values[i] > 0 then
        end
    end
end

-- EXPECTED OUTPUT:
--         return nil
--     end
--
--     return a / b
-- end
local function safe_div(a, b)
    if b == 0 then

    end
end


-- EXPECTED OUTPUT:
--             break
--         end
--         i = i + 1
--     end
--
--     return i
-- end
local function find_first_nil(values)
    local i = 1

    while i <= #values do
        if values[i] == nil then

        end
    end
end

return M

File: TestPrompts.cs

public static class TestPrompts
{
    public static int SumUpTo(int n)
    {
        int sum = 0;
        int i = 1;
        while (i <= n)
        {
        }
    }

    public static bool ContainsZero(IList<int> values)
    {
        for (int i = 0; i < values.Count; i++)
        {
            if (values[i] != 0)
            {
            }
        }
    }


    public static List<int> FilterPositive(IList<int> values)
    {
        var result = new List<int>();

        for (int i = 0; i < values.Count; i++)
        {
            if (values[i] > 0)
            {
            }
        }
    }

    public static double? SafeDiv(double a, double b)
    {
        if (b == 0)
        {
        }
    }


    public static int FindFirstNull(IList<object?> values)
    {
        int i = 0;

        while (i < values.Count)
        {
            if (values[i] == null)
            {
            }
        }
    }
}

